src/ledc.rs

```diff
@@ -503,81 +505,112 @@ mod chip {
         }
     }
 
-    /// Ledc Speed Mode
-    #[derive(PartialEq, Eq, Copy, Clone, Debug)]
-    pub enum SpeedMode {
-        #[cfg(esp_idf_soc_ledc_support_hs_mode)]
-        /// High Speed Mode. Currently only supported on the ESP32.
-        HighSpeed,
-        /// Low Speed Mode. The only configuration supported on ESP32S2, ESP32S3, ESP32C2 and ESP32C3.
-        LowSpeed,
+    /// Speed mode for the LED Control peripheral
+    /// The ESP32 supports two speed modes: low and high speed
+    /// All others support only low speed mode.
+    pub trait SpeedMode: Send + Sync + 'static {
+        const SPEED_MODE: ledc_mode_t;
     }
 
-    impl Default for SpeedMode {
-        fn default() -> Self {
-            Self::LowSpeed
-        }
+    /// Low speed mode for the LED Control peripheral
+    pub struct LowSpeed;
```

Comment:
One nit: This would mark every other LEDC peripheral on every other esp variant as only having a "LowSpeed" variant. But one could argue that its neither low nor highspeed on every other esp. So for all users expect the esp32 its a bit wired to have a LowSpeed variant, as it would imply there is another one on the used chip itself.

Reply:
But i see when using this on a esp32 that this naming scheme make sense.

Reply:
You read my mind. I had the same annoyance, but not sure how to fix, other than renaming `LowSpeed` to - say - `RegularSpeed`. 

The previous driver also named it `LowSpeed` in the `SpeedMode` enum I just retired.

So what shall it be? Leave as-is or rename to `RegularSpeed`?

Reply:
yeah, i also have no super good idea how to handle it concisely. The root cause is that we would not be generic over the LEDC Timer Driver with the extra generic term, but we would need to model as it is in reality. E.g we have two different peripheral types with "the same name". We would need to be generic over the TimerConfig. But i see that this is cumbersome since we would need two different proc-macros for impl_timer! one that is only for the esp32 and one for the rest. That way we can give them each different names.

Reply:
So yeah i don't want to block on it, either its more code if you up to it, or let it be

Reply:
I'm not sure I understand what you are suggesting. Would you please elaborate a bit?

In any case, having a _single_ driver type for a channel (`LedcDriver`) and a _single_ driver type for a timer (`LedcTimer` - even if parameterized by the peripheral type) is IMO an advantage, not a disadvantage, so ideally we should keep it that way. If that's what you suggest to break.